### PR TITLE
Expose `optimizer` and `optimizer_kwargs` in feedforward_model factory

### PR DESCRIPTION
**Pull request by ryanjdillon**
This just copies over the implementation in `lstm_model()` from `lstm_autoencoder.py` for `feedforward_autoencoder.py`. 

As such, a default optimizer is set to `adam`, with any configuration in `optimizer_kwargs` being passed to `Optim(**optimizer_kwargs)` at initialization.

This is part of the needed work to harmonize the api for these two functions, as mentioned in #283.

### CODE REVIEW

File: gordo_components/model/factories/feedforward_autoencoder.py [hunk 1]
```diff
@@ -42,6 +45,15 @@ def feedforward_model(
         Activation functions for the decoder part
     out_func: str
         Activation function for the output layer
+    optimizer: Union[str, Optimizer]
+        If str then the name of the optimizer must be provided (e.x. "adam").
+        The arguments of the optimizer can be supplied in optimization_kwargs.
```
Comment (milesgranger) [inline gordo_components/model/factories/feedforward_autoencoder.py]: `optimizer_kwargs`

File: gordo_components/model/factories/feedforward_autoencoder.py [hunk 2]
```diff
@@ -120,6 +138,15 @@ def feedforward_symmetric(
          Must have len > 0
     funcs: List[str]
         Activation functions for the internal layers
+    optimizer: Union[str, Optimizer]
+        If str then the name of the optimizer must be provided (e.x. "adam").
+        The arguments of the optimizer can be supplied in optimization_kwargs.
```
Comment (milesgranger) [inline gordo_components/model/factories/feedforward_autoencoder.py:137]: `optimizer_kwargs`

**Review by milesgranger — COMMENTED** [review]
Think it looks pretty ok; but is there not a lot of duplicate code now between these two modules?

File: gordo_components/model/factories/feedforward_autoencoder.py (hunk 1 above)
Comment (ryanjdillon) [reply gordo_components/model/factories/feedforward_autoencoder.py]: Yes indeed. I'll be back!

File: gordo_components/model/factories/feedforward_autoencoder.py [hunk 3]
```diff
@@ -86,10 +98,14 @@ def feedforward_model(
         model.add(Dense(units=units, activation=activation))
 
     # Final output layer
```
Comment (milesgranger) [inline gordo_components/model/factories/feedforward_autoencoder.py]: This comment doesn't seem relevant to this line, should probably move it below this conditional.

File: gordo_components/model/factories/feedforward_autoencoder.py [hunk 4]
```diff
@@ -86,10 +98,14 @@ def feedforward_model(
...
+    if isinstance(optimizer, str):
+        Optim = getattr(keras.optimizers, optimizer)
+        optimizer = Optim(**optimizer_kwargs)
+
     model.add(Dense(n_features_out, activation=out_func))
 
     # Set some pre-determined default compile kwargs.
-    compile_kwargs.setdefault("optimizer", "adam")
+    compile_kwargs.update({"optimizer": optimizer})
     compile_kwargs.setdefault("loss", "mean_squared_error")
```
Comment (milesgranger) [inline gordo_components/model/factories/feedforward_autoencoder.py:103]: Know you didn't write this, but maybe we should default to whatever `Keras` is (ie, not actually set a default), because that's probably where we'll refer people to documentation about optimizer and compile kwargs. If they see the default is something else, that's what I would expect it to be.

**Review by milesgranger — COMMENTED** [review]
Very nice, feel free to rebase/squash into the desired commits after addressing my small comments and can probably merge after that. :+1:

File: gordo_components/model/factories/feedforward_autoencoder.py (hunk 4 above)
Comment (ryanjdillon) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: That sounds proper to me. Should I just go ahead and do that or get a consensus from folks?

Comment (epa095) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: From [this](https://keras.io/models/sequential/) doc-string it seems like `optimizer` is a non-optional argument for model.compile, i.e. it does not have a default value. But I think we should have a default, so then adam is maybe as good a default as any?

Comment (ryanjdillon) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: @natalie-caruana @flikka Does "adam" sound good to you all here?

Comment (milesgranger) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: Sorry, I was referring to `loss` specifically; but the consensus of `optimizer` remains. :+1:

Comment (ryanjdillon) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: It looks like these are [required args](https://keras.io/losses/) for `model.compile`.

Comment (Natalie-Caruana) [reply gordo_components/model/factories/feedforward_autoencoder.py:103]: Adam optimizer as default should be fine and for the loss function mse looks good to me too.

File: gordo_components/model/factories/feedforward_autoencoder.py (hunk 1 above)
Comment (milesgranger) [reply gordo_components/model/factories/feedforward_autoencoder.py]: `optimizer_kwargs`

**Review by milesgranger — APPROVED** [review]
